onnxruntime/test/framework/nuphar/allocator_nuphar_test.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/allocatormgr.h"
+#include "core/providers/nuphar/nuphar_allocator.h"
+#include "test/framework/test_utils.h"
+
+#include "gtest/gtest.h"
+
+namespace onnxruntime {
+namespace test {
+TEST(AllocatorTest, NupharAllocatorTest) {
+  int device_id = 0;
```

Comment:
Tests in this folder are for the ORT framework library. Do nuphar specific tests belong here?

Reply:
This is following the pattern of test/framework/cuda. Any suggestions on where else it should go?

---
In reply to: [312422473](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312422473) [](ancestors = 312422473)

Reply:
Testing the allocator and copy tensor makes sense given that's implementing framework level stuff. The fusion testing seems a bit unrelated though but not sure it's worth creating onnxruntime/test/nuphar just for that so leave it for now.

---
In reply to: [312632514](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312632514) [](ancestors = 312632514,312422473)

Reply:
This file is removed as the test is covered in other places.

---
In reply to: [312762476](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312762476) [](ancestors = 312762476,312632514,312422473)